catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/CountryCodeValidator.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import com.google.common.base.Preconditions;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+
+/**
+ * Validates an attribute's value(s) against the ISO_3166-1 Alpha3 country codes.
+ * <p>
+ * Is capable of validating {@link String}s.
+ */
+public class CountryCodeValidator implements AttributeValidator {
+    private Set<String> countryCodesMap;
+
+    //TODO: Should there be an option to ignore case?
+    public CountryCodeValidator() {
+        setupISO3CountryCodes();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Validates only the values of {@code attribute} that are {@link String}s.
+     */
+    @Override
+    public Optional<AttributeValidationReport> validate(Attribute attribute) {
+        Preconditions.checkArgument(attribute != null, "The attribute cannot be null.");
+
+        final String name = attribute.getName();
+        for (final Serializable value : attribute.getValues()) {
```

Comment:
One additional thing you could do, to clean up that lambda in the middle, is extract a new private method and then invoke it with `this::newMethod`. You'd have to add an additional map call:
```java
return attribute.getValues().stream()
        .filter(String.class::isInstance)
        .map(String.class::cast)
        .filter(s -> !countryCodesMap.contains(s))
        .map(s -> name)
        .map(this::newMethod)
        .map(AttributeValidationReport.class::cast)
        .findFirst();
```

It has two advantages: firstly it makes the streaming code cleaner; secondly, it makes it more explicit that the value is being dropped on the floor after it's been determined it is not in the `countryCodesMap`.